ckan/lib/uploader.py

```diff
@@ -362,3 +362,29 @@ def download(self, id, filename=None):
             response.headers['Content-Type'] = content_type
         response.status = status
         return app_iter
+
+    def metadata(self, id, filename=None):
+        ''' Return meta details of file'''
+        try:
+            filepath = self.get_path(id)
+            content_type, content_encoding = mimetypes.guess_type(self.url)
+            hash, length = self._file_hashnlength(filepath)
+            return {'content_type': content_type, 'size': length, 'hash': hash}
+        except IOError as e:
+            logging.error("Could not retrieve meta data,  IOError thrown", e)
+            return e
```

Comment:
This isn't the expected return type.

Reply:
This is the default in other places of CKAN as it captures multiple subtypes.